### DISCUSSION TITLE
Hold CTRL to change tools "mode"

### DISCUSSION THREAD

**Discussion by novhack** [Orama-Interactive/Pixelorama#429] [Ideas]
Most tools have some binary property that has to be changed often. I propose that holding CTRL and pressing mouse button could do the opposite of selected "mode".

These options could be:
- [x] Zoom -  zoom in / zoom out
- [x] Color picker - pick for left color / pick for right
- [x] Pencil - overwrite color on / off
- [x] Bucket - fill same color area / fill same color pixels
- [x] Shading - lighten / darken option

For instance: If I have selected the lighten option and I hold CTRL I would like it to instead darken.

**Comment by ghost** [comment]
I second this - however, instead of 'CTRL', make it a defineable key, as 'CTRL' is used for copy-pasting usually, but that should go without saying.

As UI development, this could be implemented as press the 'Tool Wheel Button' and it shows the existing tools, and using left-right mouse to utilise the tools respectively.

I imagine Tablet users will thoroughly enjoy this the most - they can use Zen mode, set this as a button on their Tablet and switch tools with relative ease with enough practice. If 'Zen Mode' can be enabled by a button press, then going in to edit and out to check / alter options would be a breeze.

Not just for Tablet users, either; some people just like to edit without anything else on screen.

**Comment by OverloadedOrama** [comment]
This is now implemented in 24d25d49ead16cc820df5862d8a678ad71e8d7d1 and will be included in v0.9.1. I wanted to make this work with the Rectangle and Ellipse tools to let the users easily switch between filling the shape and not, but <kbd>Ctrl</kbd> is already reserved for centering the shape on the click origin, and there is no other free shortcut.

That being said, maybe this can be implemented once we add more customization to the shortcuts, as AtomicGerbil suggested in the above comment, so the users will be able to select some other shortcut that does not conflict. This feature is tracked in #309.
